README.md

```diff
@@ -1 +1,564 @@
-# golangcodestyle
\ No newline at end of file
+# Before you start
+
+Please take a look to the following posts:
+
+* Real world advice for writing maintainable Go programs: https://dave.cheney.net/practical-go/presentations/qcon-china.html
+This article was originally a workshop at QCon Shanghai 2018. Author gives us his advice for best practices writing Go code.
+
+* Think you understand the Single Responsibility Principle? https://hackernoon.com/you-dont-understand-the-single-responsibility-principle-abfdd005b137
+
+* Layered Architecture (and how it shows up in Microservice Architectures): https://medium.com/code-smells/layered-architecture-f11bc04c5d6c
+
+All of them are quite important for us as for backend developers.
+
+# Golang Coding Guidelines
+
+
+Based on following articles:
+
+** Effective Go - https://golang.org/doc/effective_go.html#interface-names
```

Comment:
Formatting error. Sorry, I am sensitive to such things ;) You can switch to the rendered markdown view in github, which i find quite useful to to a final check on the formatting.

Reply:
sure, fixed